src/setuptools_scm/utils.py

```diff
@@ -63,43 +84,48 @@ def ensure_stripped_str(str_or_bytes: str | bytes) -> str:
         return str_or_bytes.decode("utf-8", "surrogateescape").strip()
 
 
-def _run(cmd: _t.CMD_TYPE, cwd: _t.PathT) -> subprocess.CompletedProcess[bytes]:
+def _run(cmd: _t.CMD_TYPE, cwd: _t.PathT) -> subprocess.CompletedProcess[str]:
     return subprocess.run(
         cmd,
         capture_output=True,
         cwd=str(cwd),
         env=dict(
-            no_git_env(os.environ),
+            avoid_pip_isolation(no_git_env(os.environ)),
             # os.environ,
             # try to disable i18n
             LC_ALL="C",
             LANGUAGE="",
             HGPLAIN="1",
         ),
+        text=True,
     )
 
 
 def do_ex(cmd: _t.CMD_TYPE, cwd: _t.PathT = ".") -> _CmdResult:
-    trace("cmd", repr(cmd))
-    trace(" in", cwd)
+    if isinstance(cmd, list):
+        cmd_4_trace = " ".join(cmd)
```

Comment:
This needs to use shlex quoting to avoid commands one can't safely run copy pasted